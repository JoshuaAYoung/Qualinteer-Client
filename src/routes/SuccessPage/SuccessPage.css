.successImage {
  width: 25rem;
  margin-top: 7rem
}

.successImageContainer {
  text-align: center;
}

.successBanner {
  position: relative;
}

.successInstructions {
  font-size: 2rem;
}

@media only screen and (min-width: 1024px) {
  .successBanner {
    margin-right: 20rem;
    margin-left: 20rem;
    text-align: center;
  }
}