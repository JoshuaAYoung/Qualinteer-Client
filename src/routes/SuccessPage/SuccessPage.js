import React, { Component } from 'react';
import './SuccessPage.css';
import VableContext from '../../context/VableContext';

export default class SuccessPage extends Component {
  static contextType = VableContext;

  state = {
    redirectPath: '',
  };

  componentDidMount() {
    const params = this.props.match.params.destination;
    if (params === 'login') {
      this.setState({ redirectPath: '/login' });
    } else if (params === 'oppboard') {
      this.setState({ redirectPath: '/opportunityboard' });
    } else if (params === 'redirect') {
      this.setState({ redirectPath: '/login' });
    } else if (params === 'homepage') {
      this.setState({ redirectPath: '/' });
    } else if (params === 'logout') {
      this.setState({ redirectPath: '/' });
    }
    setTimeout(() => {
      this.props.history.push(this.state.redirectPath);
    }, 2000);
  }

  userMessage = () => {
    if (this.context.userType === 'organization') {
      return 'an organization';
    } else {
      return 'a volunteer';
    }
  };

  renderMessage = () => {
    const params = this.props.match.params.destination;
    if (params === 'login') {
      return (
        <>
          <h1 className="pageHeader">Congrats!</h1>
          <p className="pageInstructions successInstructions">
            You've successfully registered an account.
          </p>
        </>
      );
    } else if (params === 'oppboard') {
      return (
        <>
          <h1 className="pageHeader">Congrats!</h1>
          <p className="pageInstructions successInstructions">
            Your opportunity was submitted successfully.
          </p>
        </>
      );
    } else if (params === 'redirect') {
      return (
        <>
          <h1 className="pageHeader">Login</h1>
          <p className="pageInstructions successInstructions">
            Log in as a volunteer user to apply.
          </p>
        </>
      );
    } else if (params === 'homepage') {
      return (
        <>
          <h1 className="pageHeader">Congrats!</h1>
          <p className="pageInstructions successInstructions">
            You've successfully logged in as{' '}
            <span className="defaultMedium">{this.userMessage()}</span> user.
          </p>
        </>
      );
    } else if (params === 'logout') {
      return (
        <>
          <h1 className="pageHeader">Congrats!</h1>
          <p className="pageInstructions successInstructions">
            You've successfully logged out.
          </p>
        </>
      );
    }
  };

  render() {
    return (
      <div className="successBanner">
        {this.renderMessage()}
        <div className="successImageContainer">
          <img
            src="/assets/success-cloud.svg"
            alt="not found"
            className="successImage"
          />
        </div>
      </div>
    );
  }
}
