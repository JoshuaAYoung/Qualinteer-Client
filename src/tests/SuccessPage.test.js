import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SuccessPage from '../routes/SuccessPage/SuccessPage';


describe('SuccessPage Component', () => {
  const props = {
    match: {
      params: 'login'
    }
  }

  it('SuccessPage Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<SuccessPage {...props} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('SuccessPage Snapshot Test', () => {
    const wrapper = shallow(<SuccessPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})