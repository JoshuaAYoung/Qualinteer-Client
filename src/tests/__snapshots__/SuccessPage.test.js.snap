// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPage Component SuccessPage Snapshot Test 1`] = `
<div
  className="successBanner"
>
  <div
    className="successImageContainer"
  >
    <img
      alt="not found"
      className="successImage"
      src="/assets/success-cloud.svg"
    />
  </div>
</div>
`;
